ethereum/contract_addresses.go

```diff
@@ -17,8 +17,8 @@ func GetContractAddressesForChainID(chainID int) (ContractAddresses, error) {
 }
 
 func AddContractAddressesForChainID(chainID int, addresses ContractAddresses) error {
-	if _, alreadExists := ChainIDToContractAddresses[chainID]; alreadExists {
-		return fmt.Errorf("cannot add contract addresses for chain ID %d: addresses for this chain id are already defined", chainID)
+	if _, alreadExists := ChainIDToContractAddresses[chainID]; alreadExists && chainID == 1 {
+		return fmt.Errorf("cannot add contract addresses for chain ID %d: addresses for this chain id are hard-coded", chainID)
```

Comment:
Historically, this error also happened if a user tried to define contract addresses for the same `chainID` twice (e.g. via a typo in their JSON or by attempting to initialize Mesh more than once. See https://github.com/0xProject/0x-mesh/issues/619).

Why don't we change this to:

```suggestion
if chainID == 1 {
	return fmt.Errorf("cannot add contract addresses for chainID 1 : addresses for mainnet are hard-coded and cannot be changed")
}
```

This would actually fix https://github.com/0xProject/0x-mesh/issues/619, but does mean we miss potential typos in the JSON object that is passed through in the `CUSTOM_CONTRACT_ADDRESSES` environment variable, specifically for cases where the same key is specified twice. I'm okay with that change, and if we really cared about it we could validate the JSON separately.

Reply:
Yeah, I feel like the multiple keys issue is definitely an edge-case and we could add separate validation for that.